### PR TITLE
Add Java Catalog documentation.

### PR DESCRIPTION
**Pull request by davinchia**
## What
Clean up and add better guidelines on how to use the Java catalogs we recently added.

Took the chance to move existing documentation to improve reading flow.

I have deployed this so feel free to review: https://docs.airbyte.com/contributing-to-airbyte/gradle-dependency-update

## Recommended reading order
No preference.

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

### CODE REVIEW

File: docs/integrations/sources/s3.md [hunk 1]
```diff
@@ -20,12 +19,12 @@ This page contains the setup guide and reference information for the Amazon S3 s
 
 1. [Log into your Airbyte Cloud](https://cloud.airbyte.io/workspaces) account.
 2. In the left navigation bar, click **<Sources/Destinations>**. In the top-right corner, click **+new source/destination**.
-3. On the Set up the <source/destination> page, enter the name for the <connector name> connector and select **<connector name>** from the <Source/Destination> type dropdown.
```
Comment (davinchia) [inline docs/integrations/sources/s3.md:23]: this was breaking docusaurus

File: docs/integrations/sources/s3.md [hunk 2]
```diff
@@ -4,9 +4,8 @@ This page contains the setup guide and reference information for the Amazon S3 s
 
 ## Prerequisites
 
-<Connector-specific prerequisites which are required in both Airbyte Cloud & OSS>
```
Comment (davinchia) [inline docs/integrations/sources/s3.md:7]: this was also breaking docusaurus

File: docs/integrations/sources/s3.md (hunk 1 above)
Comment (supertopher) [reply docs/integrations/sources/s3.md:23]: I think the JSX translation layer is afraid of anything that vaguely resembles HTML tags

**Review by jdpgrailsdev — APPROVED** [review]
:shipit: assuming you work out the Docusaurus issues.

**Comment by davinchia** [comment]
@Amruta-Ranade I merged this in first and this is something I want to broadcast to the OSS team. Pls lmk if you have any concerns, very happy to rework/reword some of this.

**Pull request merged**